### ISSUE TITLE
Eslint v4 - no-undef not suppressed

### ISSUE THREAD

**Issue by memee** [gajus/eslint-plugin-flowtype#261]
Rules `use-flow-type` and `define-flow-type` does not suppress `no-undef` in eslint v4.3.0 with babel-eslint v7.1.1

It still works for me with eslint v3

**Comment by aaronjensen** [comment]
FWIW, I hit this by just upgrading `babel-eslint` to `7.2.3`, still on eslint `3.9.1`. `babel-eslint` `7.1.1` seems to work fine.

**Comment by memee** [comment]
I have it working with: `babel-eslint@7.2.3` and `eslint@4.3.0` and `eslint-plugin-flowtype@2.35.0`

**Comment by aaronjensen** [comment]
Thanks, that combo worked for me as well.

**Comment by linuxonrails** [comment]
It doesn't work for me (compat plugin & extends with airbnb & airbnb-flow). 

Only with babel-eslint 7.1.1.

**Comment by MoOx** [comment]
I still have issue with this code with all latest version

```
    "babel-eslint": "^7.2.3",
    "eslint": "^4.6.1",
    "eslint-plugin-flowtype": "^2.35.1",
```

```js
  handleClick = (event: SyntheticEvent<*>) => {
```

```
~/__tests__/react-class.js
  25:25  error  'SyntheticEvent' is not defined  no-undef
```

**Comment by MoOx** [comment]
It seems with recent version of eslint, babel-eslint there is new detection that cannot be magically resolved. Maybe this plugin should add some flow declaration as global var to suppress the false positives?

**Comment by MoOx** [comment]
After digging, I suspect that everything that is defined by flow (in https://github.com/facebook/flow/tree/master/lib) might have to be added as globals.

For now I am using this in my eslint config

```js
  // No need for React$* since you should use
  // import * as React from "react" and then have everything under React.* (since flow 0.53+)
  "globals": {
    "SyntheticEvent": true,
    "SyntheticAnimationEvent": true,
    "SyntheticClipboardEvent": true,
    "SyntheticCompositionEvent": true,
    "SyntheticInputEvent": true,
    "SyntheticUIEvent": true,
    "SyntheticFocusEvent": true,
    "SyntheticKeyboardEvent": true,
    "SyntheticMouseEvent": true,
    "SyntheticDragEvent": true,
    "SyntheticWheelEvent": true,
    "SyntheticTouchEvent": true,
    "SyntheticTransitionEvent": true
  }
```

**Comment by camflan** [comment]
I'm also having this issue and have resolved it using the globals solution that @MoOx provided. Is there a fix for this planned?

**Comment by dcposch** [comment]
> I have it working with: babel-eslint@7.2.3 and eslint@4.3.0 and eslint-plugin-flowtype@2.35.0

I tried that exact combination, making sure that there were no other versions of those modules in nested `node_modules`. Didn't work--still get thousands of `no-undef` errors.

I ended up not upgrading and just staying with eslint v3

**Comment by gajus** [comment]
Closing as stale.
